### PR TITLE
a few more 0.7 updates

### PR DESCRIPTION
**Pull request by stevengj**
continuing #671

### CODE REVIEW

File: src/IJulia.jl [hunk 1]
```diff
@@ -273,7 +273,8 @@ push_postexecute_hook(f::Function) = push!(postexecute_hooks, f)
 Remove a function `f()` from the list of functions to
 execute after executing any notebook cell.
 """
-pop_postexecute_hook(f::Function) = splice!(postexecute_hooks, findfirst(equalto(f), postexecute_hooks))
+pop_postexecute_hook(f::Function) =
+    splice!(postexecute_hooks, findlast(isequal(f), postexecute_hooks))
```
Comment (fredrikekre) [inline src/IJulia.jl]: `Compat.`?

File: src/IJulia.jl [hunk 2]
```diff
@@ -289,7 +290,8 @@ push_preexecute_hook(f::Function) = push!(preexecute_hooks, f)
 Remove a function `f()` from the list of functions to
 execute before executing any notebook cell.
 """
-pop_preexecute_hook(f::Function) = splice!(preexecute_hooks, findfirst(equalto(f), preexecute_hooks))
+pop_preexecute_hook(f::Function) =
+    splice!(preexecute_hooks, findlast(isequal(f), preexecute_hooks))
```
Comment (fredrikekre) [inline src/IJulia.jl]: `Compat.`?

File: src/IJulia.jl [hunk 3]
```diff
@@ -306,7 +308,8 @@ push_posterror_hook(f::Function) = push!(posterror_hooks, f)
 Remove a function `f()` from the list of functions to
 execute after an error occurs when a notebook cell is evaluated.
 """
-pop_posterror_hook(f::Function) = splice!(posterror_hooks, findfirst(equalto(f), posterror_hooks))
+pop_posterror_hook(f::Function) =
+    splice!(posterror_hooks, findlast(isequal(f), posterror_hooks))
```
Comment (fredrikekre) [inline src/IJulia.jl]: `Compat.`?

File: src/IJulia.jl (hunk 2 above)
Comment (stevengj) [reply src/IJulia.jl]: Doesn’t need it. If f isn’t found, whether findlast returns 0 or nothing splice! will throw.

**Comment by stevengj** [comment]
Looks like the Travis failure is unrelated

File: src/IJulia.jl (hunk 2 above)
Comment (stevengj) [reply src/IJulia.jl]: (But I guess it wouldn't hurt to make sure it throws the same exception on 0.6 and 0.7.)

**Pull request merged**